File: Todo-Backend/src/main/java/com/shubhamdev/todo/service/TodoService.java
package com.shubhamdev.todo.service;

import java.util.List;
import com.shubhamdev.todo.entity.Todo;

public interface TodoService {
    
    List<Todo> getAllTodos();
    
    Todo getTodoById(Integer id);
    
    Todo addTodo(Todo todo);
    
    void deleteTodo(Integer id);
    
    Todo updateTodo(Todo todo, Integer id);
}

File: Todo-Frontend/src/services/todo.service.js
// src/service/todo.service.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/todos"; // ✅ match your backend port

class TodoService {
  getAllTodos() {
    return axios.get(API_URL);
  }

  getTodoById(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  createTodo(todo) {
    return axios.post(API_URL, todo);
  }

  updateTodo(id, todo) {
    return axios.put(`${API_URL}/${id}`, todo);
  }

  deleteTodo(id) {
    return axios.delete(`${API_URL}/${id}`);
  }
}

export default new TodoService();
